### ISSUE TITLE
Issue with installing xmlstarlet on Windows

### ISSUE THREAD

**Issue by maksympashkiv** [dimitern/xmlstarlet#51]
Log in attach
Python 3.8.1
When run pip install xmlstarlet getting next error:
[error.txt](https://github.com/dimitern/xmlstarlet/files/4439199/error.txt)

**Comment by dimitern** [comment]
@maksympashkiv Hey, thanks for the report! I'll look into it on Windows.

**Comment by srolskyi** [comment]
I have the same situation. @dimitern any news?

**Comment by dimitern** [comment]
I've managed to finally setup my Windows box and reproduce the issue. Not sure of the best way to fix this though, as on Linux/Mac libxml headers and sources are easy enough to install using the system package management CLI. On Windows if the sources of libxml are put in the src directory of a checkout build seems to pass fine, but requires manual steps.

**Comment by maksympashkiv** [comment]
> I've managed to finally setup my Windows box and reproduce the issue. Not sure of the best way to fix this though, as on Linux/Mac libxml headers and sources are easy enough to install using the system package management CLI. On Windows if the sources of libxml are put in the src directory of a checkout build seems to pass fine, but requires manual steps.

Did you test from this repo? Because I get next error:
C:\Users\Administrator\Desktop\Test\xmlstarlet\xmlstarlet\src\xmlstar.h(4): fatal error C1083: Cannot open include file: 'config.h': No such file or directory

In this repo I can't find that file...Mb I doing something wrong? libxml sources i put in :)

**Comment by maksympashkiv** [comment]
@dimitern Hi! Maybe you can share your sources for windows? Or write short instructions about installation? I would really appreciate it.
Thx.

**Comment by dimitern** [comment]
So that config.h is generated on *NIX systems by the configure script, it's system-dependent and cannot be checked into the repo. As for documentation, try the xmlstarlet docs on sf.net (http://xmlstar.sourceforge.net/doc/UG/ch02s04.html) not more than a single line I'm afraid. If you can use WSL or Cygwin you should be able to build it fine on Windows. Hope it helps.

**Comment by dimitern** [comment]
I realize a native Windows binary wheel would be necessary to make the installation possible, so tomorrow I'll work on this and if it goes well I'll release it to pypi when done.

**Comment by dimitern** [comment]
I managed to build the package on Windows after enabling Windows Subsystem for Linux fine, but I'm still not sure how to make a native binary wheel for Windows. Even the original xmlstarlet project seem to suffer from the same issues (Cygwin / Mingw / Msys is needed to build an .exe, and there is only an older binary build available for download). So I can explain how I did it, however I'm not sure if it will be helpful in your respective cases.

This is what I did:
1. Followed the guide to enable WSL: https://docs.microsoft.com/en-us/windows/wsl/install-win10
2. Installed Ubuntu 18.04 from Windows Store
3. Inside it, I ran `sudo apt update -qyy && sudo apt full-upgrade -yqq && sudo apt install build-essential git python3 python3-pip libxml2-dev autoconf tox`
4. Cloned this repo inside the Ubuntu shell: `git clone https://github.com/dimitern/xmlstarlet.git'
5. Created a virutalenv with the packages: `cd xmlstarlet && python3 -mvenv .venv && . .venv/bin/activate && pip install -r requirements.txt`
6. Finally, ran `python setup.py sdist bdist_wheel`.

The binary wheel is in `dist/` dir, but it is marked as a Linux wheel, not sure if it can be used outside WSL shell (probably not).

I hope at least some of this helps.
